Dockerfile

```diff
@@ -1,18 +1,20 @@
-FROM ocaml/opam:debian-12-ocaml-4.14
+FROM ocaml/opam:debian-12-ocaml-5.2
 RUN sudo apt-get update && sudo apt-get install autoconf automake -y --no-install-recommends
 RUN mkdir -p /home/opam/www/mirage
 WORKDIR /home/opam/www
-RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam && cd ~/opam-repository && git pull origin master && git reset --hard 297ab9ebb38254eee222f5bb4365ae80baa9caa1 && opam update
-RUN opam pin git+https://github.com/tmattio/opam-tailwindcss#a4c0ee9b9a80d44c4a953b07dc29e4aa065adf57
-RUN opam repo add mirage-dev git+https://github.com/mirage/mirage-dev.git#842c55556ffd0950d21141d6ab99e52a8d88a50f
-RUN opam install mirage
+RUN sudo ln -f /usr/bin/opam-2.1 /usr/bin/opam
+RUN cd ~/opam-repository && git pull origin master && git reset --hard 705cfbfe709bba2533a3cd6164b7dab37fca0342
+RUN opam update
+RUN opam install 'mirage>=4.5.0'
 COPY --chown=opam:root mirage/config.ml /home/opam/www/mirage/
 COPY --chown=opam:root mirageio.opam /home/opam/www/
 ARG TARGET=unix
 ARG EXTRA_FLAGS=
-RUN opam exec -- mirage configure -f mirage/config.ml -t $TARGET $EXTRA_FLAGS
+ARG EXTRA_FLAGS_NO_METRICS="--tls=true --separate-networks"
```

Comment:
I wondered why not use `EXTRA_FLAGS` instead of creating the `EXTRA_FLAGS_NO_METRICS`.
Or, taken the other way around, if keeping `EXTRA_FLAGS` separate is useful, shouldn’t it still appear in the configuration command lines?

Reply:
Basically, the CI system always passes `--metrics` through `EXTRA_FLAGS` when building `mirage-www` (and I don't have control over that CI system). And I couldn't find a reasonable way to do something like `EXTRA_FLAGS_NO_METRICS = $EXTRA_FLAGS without --metrics`.

Another way would be for this branch to still accept --metrics but disable the underlying behavior (and not pull the dependency).

I went with the hacky solution (to hardcode what the CI passes without --metrics) to modify the code as little as possible so this experiment is representative. Then the correct solution will be to change the CI system to pass the correct flags, once we know we want to do this on `main`.